projects/wave-core-new/src/lib/colors/color-map-selector/color-map-selector.component.ts

```diff
@@ -0,0 +1,229 @@
+            if (Number.isFinite(x.min)) {
+                this.minValueChange.emit(x.min.value);
+            }
+            if (Number.isFinite(x.max)) {
+                this.maxValueChange.emit(x.max.value);
+            }
+        });
+        this.subscriptions.push(subMinMax);
+    }
+
+    ngOnDestroy(): void {
+        this.subscriptions.forEach((s) => s.unsubscribe());
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.minValue || changes.maxValue) {
+            this.patchMinMaxValues(this.minValue, this.maxValue);
+        }
+    }
+
+    /**
+     * Replace the min and max values.
+     */
+    patchMinMaxValues(min: number, max: number): void {
+        if (typeof min !== 'number' || typeof max !== 'number') {
+            return;
+        }
+
+        const bounds: {min: number; max: number} = this.form.controls['bounds'].value;
+
+        if (bounds.min === min && bounds.max === max) {
+            return;
+        }
+
+        this.form.controls.bounds.setValue({min, max});
+
+        this.updateColorizerData();
+    }
+
+    /**
+     * Clears the local colorizer data.
+     */
+    removeColorizerData(): void {
+        this.breakpoints = undefined;
+    }
+
+    /**
+     * Apply a new color table to the colorizer data.
+     */
+    applyNewColorTable(_: any): void {
+        if (!this.breakpoints) {
+            return;
+        }
+
+        this.breakpointsChange.emit(this.breakpoints);
+    }
+
+    protected updateColorizerData(): void {
+        if (!this.checkValidConfig()) {
+            this.breakpoints = undefined;
+            return;
+        }
+
+        this.breakpoints = this.createBreakpoints();
+
+        this.changeDetectorRef.markForCheck();
+    }
+
+    protected checkValidConfig(): boolean {
+        const colorMap: Array<RgbaTuple> = this.form.controls['colorMap'].value;
+        const colorMapSteps: number = this.form.controls['colorMapSteps'].value;
+        const boundsMin: number = this.form.controls['bounds'].value.min;
+        const boundsMax: number = this.form.controls['bounds'].value.max;
+
+        if (!colorMap) {
+            return false;
+        }
+        if (colorMapSteps > this.maxColormapSteps) {
+            return false;
+        }
+        if (boundsMin >= boundsMax) {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected createBreakpoints(): Array<ColorBreakpoint> {
+        const colorMap: Array<RgbaTuple> = this.form.controls['colorMap'].value;
+        const colorMapSteps: number = this.form.controls['colorMapSteps'].value;
+        const colorMapReverseColors: boolean = this.form.controls['colorMapReverseColors'].value;
+        const bounds: {min: number; max: number} = this.form.controls['bounds'].value;
+
+        // TODO: allow different gradient types
+
+        const breakpoints = new Array<ColorBreakpoint>();
+
+        const stepSize = 1 / (colorMapSteps - 1);
```

Comment:
we coould add a test for this logic ;)

Reply:
We should, but we need to first initialize tests.